### ISSUE TITLE
KVM_CAP_DIRTY_LOG_RING not supported, QEMU-Nyx fails with "Assertion `ret < 0' failed

### ISSUE THREAD

**Issue by DMIND-NLL** [nyx-fuzz/packer#30]
Hello,

I am encountering an issue when trying to run the installation process using qemu_tool.sh for a VM image with QEMU-Nyx. I attempted to install the image as per the instructions, but I keep running into an error related to KVM_CAP_DIRTY_LOG_RING in fallback mode. Here are the details of my environment and the error output:

Environment:

OS: Ubuntu 18.04.6 (Live Server ISO)
QEMU-Nyx : As per the setup instructions in the QEMU-Nyx repository
Host: Ubuntu 22.04
Kernel: 5.10.75-051075-generic (as per the setup instructions in the KVM-Nyx repository)
KVM module: Installed and enabled (kvm and kvm-intel)

Command:
```
../packer/qemu_tool.sh install ubuntu_docker.img ubuntu-18.04.6-live-server-amd64.iso
```

Error Output:
```
INSTALL
[QEMU-NYX] Warning: Nyx block COW layer disabled for ubuntu_docker.img (write operations are not cached!)
[QEMU-NYX] Warning: Nyx block COW layer disabled for ubuntu-18.04.6-live-server-amd64.iso (write operations are not cached!)
[QEMU-Nyx] Could not access KVM-PT kernel module!
[QEMU-Nyx] Trying vanilla KVM...
[QEMU-Nyx] NYX runs in fallback mode (no Intel-PT tracing or nested hypercall support)!
[QEMU-Nyx] Error: NYX requires support for KVM_CAP_DIRTY_LOG_RING in fallback mode!
qemu-system-x86_64: /home/ubuntu/QEMU-Nyx/accel/kvm/kvm-all.c:2238: kvm_init: Assertion `ret < 0' failed.
../packer/qemu_tool.sh：行 85:  3842 已放弃               （核心已转储） NYX_DISABLE_DIRTY_RING=y NYX_DISABLE_BLOCK_COW=TRUE $QEMU_PT_BIN --enable-kvm -drive format=raw,file=$2 -cdrom $3 -k de -vnc :0 -m $MEMORY_SIZE
```
![image](https://github.com/user-attachments/assets/e651f524-2955-4c2d-9ec4-7af39b0ae960)

Additional Information:

- I have followed the Kernel setup instructions provided in the KVM-Nyx documentation and am using Kernel 5.10.75-051075-generic.
- The KVM modules (kvm and kvm-intel) are installed and active on the host system.

Could you provide guidance on enabling support for KVM_CAP_DIRTY_LOG_RING in fallback mode, or any suggestions on how to address this issue?

Thank you for your assistance and for the great work on this project.

**Comment by Fewword** [comment]
Hello, did you solved it? @DMIND-NLL

**Comment by DMIND-NLL** [comment]
@Fewword 
No, It’s still puzzling me. 😖

**Comment by Fewword** [comment]
@DMIND-NLL After investigating the issue with running the qemu_tool.sh script, I found that the problem was caused by using the latest version of QEMU-Nyx, which is not compatible with the script. Switching to the master branch of QEMU-Nyx resolved the problem.

**Comment by DMIND-NLL** [comment]
@Fewword 
Thanks a lot for your help! 👍🙂

**Comment by Fewword** [comment]
@DMIND-NLL I am currently encountering a new issue. Although the master version of qemu-nyx can successfully install and postinstall, it cannot createe_snapshot because the master version does not support the fast-vm-load option. This option is only available in qemu-nyx-4.2.0, so the previous issue is happening again.
